core/trino-main/src/main/java/io/trino/operator/aggregation/state/LongDecimalWithOverflowStateFactory.java

```diff
@@ -133,6 +141,12 @@ public void setOverflow(long overflow)
             this.overflow = overflow;
         }
 
+        @Override
+        public void addOverflow(long overflow)
+        {
```

Comment:
why no `if (overflow != 0) {` check?

Reply:
Because the addition is cheap. In the grouped case, we guard it because the cause to read/set the value in the underlying array is more expensive